src/node/SetupServerCommonApi.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * @note This API is extended by both "msw/node" and "msw/native"
+ * so be minding about the things you import!
+ */
+import type { RequiredDeep } from 'type-fest'
+import { invariant } from 'outvariant'
+import {
+  BatchInterceptor,
+  InterceptorReadyState,
+  type HttpRequestEventMap,
+  type Interceptor,
+} from '@mswjs/interceptors'
+import type { LifeCycleEventsMap, SharedOptions } from '~/core/sharedOptions'
+import { SetupApi } from '~/core/SetupApi'
+import { handleRequest } from '~/core/utils/handleRequest'
+import type { RequestHandler } from '~/core/handlers/RequestHandler'
+import { mergeRight } from '~/core/utils/internal/mergeRight'
+import { devUtils } from '~/core/utils/internal/devUtils'
+import type { ListenOptions, SetupServerCommon } from './glossary'
+import type { Socket } from 'socket.io-client'
+import {
+  SyncServerEventsMap,
+  createSyncClient,
+  createWebSocketServerUrl,
+} from './setupRemoteServer'
+import {
+  onAnyEvent,
+  serializeEventPayload,
+} from '~/core/utils/internal/emitterUtils'
+import { RemoteRequestHandler } from '~/core/handlers/RemoteRequestHandler'
+
+export const DEFAULT_LISTEN_OPTIONS: RequiredDeep<SharedOptions> = {
+  onUnhandledRequest: 'warn',
+}
+
+interface InteractiveRequest extends Request {
+  respondWith(response: Response): void
+}
+
+export class SetupServerCommonApi
+  extends SetupApi<LifeCycleEventsMap>
+  implements SetupServerCommon
+{
+  protected readonly interceptor: BatchInterceptor<
+    Array<Interceptor<HttpRequestEventMap>>,
+    HttpRequestEventMap
+  >
+  private resolvedOptions: RequiredDeep<ListenOptions>
+  private socket: Socket<SyncServerEventsMap> | undefined
+
+  constructor(
+    interceptors: Array<{ new (): Interceptor<HttpRequestEventMap> }>,
+    handlers: Array<RequestHandler>,
+  ) {
+    super(...handlers)
+
+    this.interceptor = new BatchInterceptor({
+      name: 'setup-server',
+      interceptors: interceptors.map((Interceptor) => new Interceptor()),
+    })
+
+    this.resolvedOptions = {} as RequiredDeep<ListenOptions>
+
+    this.init()
+  }
+
+  /**
+   * Subscribe to all requests that are using the interceptor object
+   */
+  private init(): void {
+    this.interceptor.on('request', async ({ request, requestId }) => {
+      // If in remote mode, await the WebSocket connection promise
+      // before handling any requests.
+      if (this.resolvedOptions.remotePort) {
+        const remoteResolutionUrl = createWebSocketServerUrl(
+          this.resolvedOptions.remotePort,
+        )
+        // Build a pattern to match the socket.io connection
+        remoteResolutionUrl.pathname = '/socket.io'
+        const matcher = new RegExp(`${remoteResolutionUrl.href}/.*`)
+        const isRequestForSocket = matcher.test(request.url)
+        // Bypass handling if the request is for the socket.io connection
+        if (isRequestForSocket) {
+          return
+        }
```

Comment:
The pathname here depends on socket-io's default value, because MSW doesn't set the value in opts.path (see setupRemoteServer.ts, line 210, second arg).
If you preferred, then that location could be changed to set the path, and then this wouldn't be dependent on their potential changes to default values.